### PR TITLE
Updated module for new mail setting configuration

### PR DESCRIPTION
**Pull request by siddharthashw**
Fixes - #215 & #225 & #226 

1. Client.cs - updated mail sending functionality
2. IMailSettingDetailsByProjectAndModule.cs - new repository to get mail setting details by module wise
3. MailSettingDetailsByProjectAndModule.cs - new repository to get mail setting details by module wise
4. IOauthCallsRepository.cs - added new method GetListOfProjectsEnrollmentOfUserByUserId
5. OauthCallsRepository.cs - added new method GetListOfProjectsEnrollmentOfUserByUserId
6. SlackRepository.cs - updated to mail sending functionality
7. TaskMailRepository.cs - updated to mail sending functionality
8. EmailApplication.cs - updated To from string to List of string and added new property CC
9. EmailService.cs - updated mail sending functionality
10. AutofacConfig.cs - register IMailSettingDetailsByProjectAndModule and MailSettingDetailsByProjectAndModule
11. mailsetting.html - resolve issue in auto-selected-chip
12. AutofacConfig.cs (Test) - register IMailSettingDetailsByProjectAndModuleRepository
13. MailSettingDetailsByProjectAndModuleTest.cs - added test for IMailSettingDetailsByProjectAndModuleRepository
14. OauthcallsRespositoryTest.cs - added test cases for new method
15. SlackRepositoryTest.cs - updated the test cases as per new mailsetting
16. MailSettingRepository.cs - refactor the repository
17. Index.html - added path for logout

### CODE REVIEW

File: Slack.Automation/Promact.Core.Repository/Client/Client.cs [hunk 1]
```diff
@@ -143,13 +147,21 @@ public class Client : IClient
...
             var teamLeaders = await _oauthCallRepository.GetTeamLeaderUserIdAsync(userId, accessToken);
             var management = await _oauthCallRepository.GetManagementUserNameAsync(accessToken);
             var userDetail = await _oauthCallRepository.GetUserByUserIdAsync(userId, accessToken);
-            foreach (var user in management)
-            {
-                teamLeaders.Add(user);
-            }
             foreach (var teamLeader in teamLeaders)
             {
                 var user = await _userManager.FindByIdAsync(teamLeader.Id);
```
Comment (ankit213) [inline Slack.Automation/Promact.Core.Repository/Client/Client.cs]: I did not get it. Where are you using this user and teamLeader variable ?

File: Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            }
+            return mailSetting;
+        }
+
+        /// <summary>
+        /// Method used to delete duplicate string in a list
+        /// </summary>
+        /// <param name="listOfString">list of string</param>
+        /// <returns>list of string</returns>
+        public List<string> DeleteTheDuplicateString(List<string> listOfString)
```
Comment (ankit213) [inline Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs]: method must be optimized

File: Slack.Automation/Promact.Core.Repository/Client/Client.cs (hunk 1 above)
Comment (siddharthashw) [reply Slack.Automation/Promact.Core.Repository/Client/Client.cs]: updated

File: Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs (hunk 2 above)
Comment (siddharthashw) [reply Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs]: removed from repository

File: Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        /// Method used to get mail setting for a module by projectId
+        /// </summary>
+        /// <param name="projectId">project Id</param>
+        /// <param name="module">mail setting module</param>
+        /// <param name="userId">user's user Id</param>
+        /// <returns>mail setting details</returns>
+        public async Task<MailSettingAC> GetMailSettingAsync(int projectId, string module, string userId)
+        {
+            MailSettingAC mailSetting = new MailSettingAC();
+            mailSetting.To = new List<string>();
```
Comment (ankit213) [inline Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs:51]: Why are you assigning empty list of string?

File: Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            mailSetting.CC = new List<string>();
```
Comment (ankit213) [inline Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs:52]: same as above

File: Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs [hunk 5]
```diff
@@ -301,23 +308,26 @@ public class TaskMailRepository : ITaskMailRepository
                                             case SendEmailConfirmation.yes:
                                                 {
                                                     EmailApplication email = new EmailApplication();
+                                                    email.To = new List<string>();
```
Comment (ankit213) [inline Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs:311]: same as above

File: Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs [hunk 6]
```diff
@@ -301,23 +308,26 @@ public class TaskMailRepository : ITaskMailRepository
...
+                                                    email.CC = new List<string>();
```
Comment (ankit213) [inline Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs:312]: same as above

File: Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs (hunk 3 above)
Comment (siddharthashw) [reply Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs:51]: this is initialize....else I will not be able to add string in list

File: Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs (hunk 4 above)
Comment (siddharthashw) [reply Slack.Automation/Promact.Core.Repository/MailSettingDetailsByProjectAndModule/MailSettingDetailsByProjectAndModuleRepository.cs:52]: this is initialize....else I will not be able to add string in list

File: Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs (hunk 5 above)
Comment (siddharthashw) [reply Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs:311]: this is initialize....else I will not be able to add string in list

File: Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs (hunk 6 above)
Comment (siddharthashw) [reply Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs:312]: this is initialize....else I will not be able to add string in list

**Pull request merged**